examples/simulation_pytorch/pytorch_simulation.py

```diff
@@ -0,0 +1,129 @@
+    strategy = FedAvg(min_available_clients=num_clients, fraction_fit=fraction_fit)
+    # Exposes the server by default on port 8080
+    fl.server.start_server(strategy=strategy, config={"num_rounds": num_rounds})
+
+
+def start_client(pytorch_dataloader: DATASET) -> None:
+    """Start a single client with the provided dataset."""
+
+    # Load and compile a Keras model for CIFAR-10
+    net = SimpleNet().to(DEVICE)
+
+    # Unpack the CIFAR-10 dataset partition
+    trainloader, testloader = pytorch_dataloader
+
+    class CifarClient(fl.client.NumPyClient):
+        def get_parameters(self):
+            return [val.cpu().numpy() for _, val in net.state_dict().items()]
+
+        def set_parameters(self, parameters):
+            params_dict = zip(net.state_dict().keys(), parameters)
+            state_dict = OrderedDict({k: torch.Tensor(v) for k, v in params_dict})
+            net.load_state_dict(state_dict, strict=True)
+
+        def fit(self, parameters, config):
+            self.set_parameters(parameters)
+            train(net, trainloader, epochs=1)
+            return self.get_parameters(), len(trainloader), {}
+
+        def evaluate(self, parameters, config):
+            self.set_parameters(parameters)
+            loss, accuracy = test(net, testloader)
+            return float(loss), len(testloader), {"accuracy": float(accuracy)}
+
+    # Start Flower client
+    fl.client.start_numpy_client("0.0.0.0:8080", client=CifarClient())
+
+
+def train(net, trainloader, epochs):
+    """Train the network on the training set."""
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.SGD(net.parameters(), lr=0.001, momentum=0.9)
+    net.train()
+    for _ in range(epochs):
+        for images, labels in trainloader:
+            images, labels = images.to(DEVICE), labels.to(DEVICE)
+            optimizer.zero_grad()
+            loss = criterion(net(images), labels)
+            loss.backward()
+            optimizer.step()
+
+
+def test(net, testloader):
+    """Validate the network on the entire test set."""
+    criterion = torch.nn.CrossEntropyLoss()
+    correct, total, loss = 0, 0, 0.0
+    net.eval()
+    with torch.no_grad():
+        for data in testloader:
+            images, labels = data[0].to(DEVICE), data[1].to(DEVICE)
+            outputs = net(images)
+            loss += criterion(outputs, labels).item()
+            _, predicted = torch.max(outputs.data, 1)
+            total += labels.size(0)
+            correct += (predicted == labels).sum().item()
+    accuracy = correct / total
+    return loss, accuracy
+
+
+def run_simulation(num_rounds: int, num_clients: int, fraction_fit: float):
+    """Start a FL simulation."""
+    logging.basicConfig(
+        filename="pytorch_simulation.log",
+        filemode="w",
+        level=logging.DEBUG,
+        format="%(asctime)s %(levelname)s: %(message)s",
+        datefmt="%d-%m-%Y %I:%M:%S %p",
+    )
```

Comment:
```suggestion
        fl.common.logger.configure("simulation", "simulation.log")
```

Reply:
@MakGulati I tried running your example and it gets stuck after the sampling. Would it be possible for you to also add a Dockerfile and `run.sh` script similar to the other simulation example in which the training finishes? I want to make the issue lies is not because of different environment settings.

Reply:
@tanertopal I updated the code as per your advice but I am still facing some issues(memory overflow) while running docker container. I was using conda environment on my machine. Could you please try this conda environment settings from my [environment.yml ](https://gist.github.com/MakGulati/4f32a98924adbee467332387f644b21b).
Pleas let me know if there is still some problem. Meanwhile I will check what I can do with the docker file.

Reply:
@tanertopal Please see the updates. I hope I covered all the requirements.
Thank you :)

Reply:
@MakGulati amazing thank you too! I will try it out soon and provide you with feedback.